src/BinSkim.Rules/PERules/BA2006.BuildWithSecureTools.cs

```diff
@@ -258,33 +210,23 @@
 
                 if (foundIssue)
                 {
+                    if (!languageToBadModules.TryGetValue(omDetails.Language, out List<ObjectModuleDetails> badModules))
+                    {
+                        badModules = new List<ObjectModuleDetails>();
+                    }
+
                     badModules.Add(omDetails);
+                    languageToBadModules[omDetails.Language] = badModules;
                 }
                 else
                 {
                     goodCompilers.Add(BuildCompilerIdentifier(omDetails));
```

Comment:
>goodCompilers [](http://example.com/codeflow?start=20&length=13)

While we're here can you please change this variable name to 'inPolicyCompilers'? This is more descriptive than 'good'.